src/main/java/net/coderbot/iris/gui/property/Property.java

```diff
@@ -0,0 +1,46 @@
+package net.coderbot.iris.gui.property;
+
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.font.TextRenderer;
+import net.minecraft.client.util.math.MatrixStack;
+import net.minecraft.text.LiteralText;
+import net.minecraft.text.Text;
+
+/**
+ * An element of a document. Used for options
+ * or configuration menus. Property objects are
+ * added to a PropertyList, and PropertyLists are
+ * added to a PropertyDocumentWidget which is used
+ * in a GUI.
+ */
+public class Property {
```

Comment:
This seems like something that should be abstract?

Reply:
Sure. There aren't any abstract methods to inherit, but it makes more sense as it's a base class.

Reply:
I have decided not to make the class abstract since it may imply that it is not meant to be instantiated, which it is meant to be.